projects/core/dsl/runtime.py

```
@@ -142,8 +176,20 @@ def _execute_single_task(task_info, args):
+
+        # Wrap in custom exception with context
+        task_error = TaskExecutionError(
+            task_name=task_name,
+            task_description=task_func.__doc__ or 'No description',
+            original_exception=e,
+            task_args=vars(args) if hasattr(args, '__dict__') else None,
+            task_location=task_location,
+            artifact_dir=str(env.ARTIFACT_DIR),
+        )
+        # Don't use 'from None' - preserve the original stack trace
+        raise task_error
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)